src/ui/components/label.c

```diff
@@ -42,7 +42,43 @@ typedef struct {
 void label_update(component_t* component, const char* text)
 {
     data_t* data = (data_t*)component->data;
+    component_t* parent = component->parent;
     snprintf(data->text, sizeof(data->text), "%s", text);
+    UG_MeasureString(&(component->dimension.width), &(component->dimension.height), text);
```

Comment:
i'd add a check here that parent != NULL and return if it is NULL. i am not confident we set the parent diligently everywhere.

Reply:
ack